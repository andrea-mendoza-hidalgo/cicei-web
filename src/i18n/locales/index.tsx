import en from './translations.en.json';
import es from './translations.es.json';
import { SharedEn, SharedEs } from '../../shared/components/locale';
import { PagesEn, PagesEs } from '../../components/pages/locales';
import ProjectEn from '../../components/project/locale/project.en.json';
import ProjectEs from '../../components/project/locale/project.es.json';
import PublicationsEn from '../../components/publication/locale/publication.en.json';
import PublicationsEs from '../../components/publication/locale/publication.es.json';
import ResearchersEn from '../../components/researchers/locale/researchers.en.json';
import ResearchersEs from '../../components/researchers/locale/researchers.es.json';
import NewsEs from '../../components/news/locale/news.es.json';
import NewsEn from '../../components/news/locale/news.en.json';

en.shared= SharedEn;
es.shared= SharedEs;
en.pages = PagesEn;
es.pages = PagesEs;
en.project = ProjectEn;
es.project = ProjectEs;
en.publication = PublicationsEn;
es.publication = PublicationsEs;
en.researchers = ResearchersEn;
es.researchers = ResearchersEs;
en.news = NewsEn;
es.news = NewsEs;



export {en, es};