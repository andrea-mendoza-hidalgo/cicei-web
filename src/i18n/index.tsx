import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import { es, en } from './locales';

const options = {
  debug: false,

  resources: {
    'es-ES': {
      shared: es.shared,
      pages: es.pages,
      project: es.project,
      publication: es.publication,
      researchers: es.researchers,
      news: es.news
    },
    'en-US': {
      shared: en.shared,
      pages: en.pages,
      project: en.project,
      publication: en.publication,
      researchers: en.researchers,
      news: en.news
    }
  },

  fallbackLng: 'es-ES',

  ns: ['shared', 'pages', 'project', 'publication'],

  defaultNS: 'shared',

  react: {
    wait: true
  }
};

i18n.use(initReactI18next).use(LanguageDetector).init(options);

export default i18n;
