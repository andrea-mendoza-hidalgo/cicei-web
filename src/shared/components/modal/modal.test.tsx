import React from 'react';
import { mount, shallow } from 'enzyme';
import { Link, MemoryRouter } from 'react-router-dom';
import '../../../i18n';
import { fireEvent, render, waitForElement, within } from '@testing-library/react';
import CustomModal from './modal';

const modalData = {
  _id: '5fc119e64748900603413058',
  projectCode: 'C3PO',
  projectTitle: 'Proyecto',
  projectInvestigator: '',
  projectStudent: 'Carlos Arias',
  projectDateStart: '2020-11-27',
  projectDateEnd: '2021-02-18',
  projectInstitution: 'Universidad Mayor de San Simon',
  projectFunding: 'Centro de InvestigaciÃ³n de Ciencias Exactas e Ingenieria',
  projectBudget: '0',
  projectCurrency: 'Boliviano',
  projectProgress: '30',
  projectAbstract:
    'lorem ipsum asdasd asdadjasd lorem ipsum asdasd asdadjasd lorem ipsum asdasd asdadjasd lorem ipsum asdasd asdadjasd lorem ipsum asdasd asdadjasd lorem ipsum asdasd asdadjasd lorem ipsum asdasd asdadjasd lorem ipsum asdasd asdadjasd ',
  userOwner: '5fbf18465e2d8d00179f4645',
  __v: 0
};

const modalDetails = [
    { key: 'projectCode', label: 'code' },
    { key: 'projectTitle', label: 'title' },
    { key: 'projectInvestigator', label: 'investigator' },
    { key: 'projectInstitution', label: 'institution' },
    { key: 'projectStudent', label: 'collaborator'},
    { key: 'projectDateStart', label: 'dateStart'},
    { key: 'projectAbstract', label: 'abstract'}
  ];


describe('Modal Render Test', () => {
  it('Modal renders correctly ', () => {
    const handleModalOpen = jest.fn();

    let Modal = mount(
      <MemoryRouter>
        <CustomModal data={modalData} keys={modalDetails} state={true} title={'Proyecto'} handleClickClose={handleModalOpen} />
      </MemoryRouter>
    );
    
    expect(Modal).toMatchSnapshot();
  });

  it('Modal call handleClosed method correctly ', () => {
    const handleModalClosed = jest.fn();

    let Modal = mount(
      <MemoryRouter>
        <CustomModal data={modalData} keys={modalDetails} state={true} title={'Proyecto'} handleClickClose={handleModalClosed} />
      </MemoryRouter>
    );


    Modal.find('button').simulate('click');
    
    expect(handleModalClosed).toBeCalled();
  });
});
