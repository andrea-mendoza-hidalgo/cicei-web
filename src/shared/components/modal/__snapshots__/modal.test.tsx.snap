// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Render Test Modal renders correctly  1`] = `ReactWrapper {}`;
